lib/adapters/class-vip-enterprise-search.php

```diff
@@ -18,15 +19,109 @@
  */
 class VIP_Enterprise_Search extends Adapter {
 
+	/**
+	 * Add facets to EP query.
+	 * Filters `ep_post_formatted_args`.
+	 *
+	 * @param array $query
+	 * @return array
+	 * @see \ElasticPress\Indexable\Post\Post.
+	 */
+	public function add_facets_to_ep_query( $query ) {
+		// Do we have any facets specified?
+		$searched_facets = get_query_var( 'fs' );
+		if ( empty( $searched_facets ) ) {
+			return $query;
+		}
+
+		// Try to get the list of configured facets.
+		$configured_facets = $this->get_facet_config();
+		if ( empty( $configured_facets ) ) {
+			return $query;
+		}
+
+		// Match searched facets against configured facets and augment the query accordingly.
+		foreach ( $searched_facets as $facet_label => $facet_terms ) {
+			// Skip any specified facets that are not configured.
+			if ( empty( $configured_facets[ $facet_label ]['type'] ) ) {
+				continue;
+			}
+
+			// Loop over terms and add each based on type.
+			foreach ( $facet_terms as $facet_term ) { // phpcs:ignore WordPressVIPMinimum.Variables.VariableAnalysis.UnusedVariable
+				switch ( $configured_facets[ $facet_label ]['type'] ) {
+					case 'taxonomy':
+						$query['query']['function_score']['query']['bool']['must'][] = [
+							'terms' => [
+								'terms.' . $configured_facets[ $facet_label ]['taxonomy'] . '.slug' => [
```

Comment:
Is it possible or easy to also map via `term_id` as well as `slug`? I think the current VIP Search widget has a filter that allows for that.

Reply:
I'm not sure I follow - what would we gain by adding an additional clause here?

Reply:
It's my understanding it is another option. You could match by the `term_id` instead of the term `slug`.

Reply:
This might be useful in combination with the suggestion above about how the data makes its way to this plugin for applying the facets. On a headless site, it might use the term ID instead of the term slug. However, the idea behind this plugin is that it handles faceting for you, so being opinionated about how it does that is fine. If we discover that we need additional flexibility around accepting facet configurations via REST or GraphQL we could revisit the idea of allowing this to be a term ID instead of a term slug.